python/modules/qcdSmearProducer.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+import os, sys
+import ROOT
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+from importlib import import_module
+
+from PhysicsTools.NanoAODTools.postprocessing.framework.datamodel import Collection, Object
+from PhysicsTools.NanoAODTools.postprocessing.framework.eventloop import Module
+from PhysicsTools.NanoAODTools.postprocessing.tools import deltaPhi, deltaR, closest
+
+class qcdSmearProducer(Module): 
+    def __init__(self):
+	self.writeHistFile=True
+	self.metBranchName="MET"
+ 
+    def beginJob(self,histFile=None,histDirName=None):
+   	pass
+    def endJob(self):
+	pass 
+
+    def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        self.out = wrappedOutputTree
+	self.out.branch("origRes", "F", lenVar="nJet*nGenJet");
+	self.out.branch("jetFlav", "F");
+
+    def endFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
+        pass
+
+    def jetResFunction(self, jets, genjets):
+	res = jets.pt/genjets.pt
```

Comment:
So this is the jet response? It seems you are calculate each jet with each genjet? No need for matching?

Reply:
I agree there should be a matching of the reco and genjet before this value is calculated. I only included the function as an example for Dipika

Reply:
I see. Actually NanoAOD stored the index to genjet  as
"Jet_genJetIdx | Int_t(index to Genjet) | index of matched gen jet"

You can just call that for linking to genjet.